src/app/shared/horizontal-carousel/horizontal-carousel.ts

```diff
@@ -0,0 +1,88 @@
+import {AfterViewInit, Component, NgModule, ViewEncapsulation} from '@angular/core';
+import {GuideItems} from '../guide-items/guide-items';
+import {MatCardModule} from '@angular/material/card';
+import {CommonModule} from '@angular/common';
+import {RouterModule} from '@angular/router';
+import {MatIconModule} from '@angular/material/icon';
+import {MatButtonModule} from '@angular/material/button';
+
+const CARD_WIDTH = 250;
+
+@Component({
+  selector: 'app-horizontal-carousel',
+  templateUrl: './horizontal-carousel.html',
+  styleUrls: ['./horizontal-carousel.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+
+export class HorizontalCarousel implements AfterViewInit {
+  position: number = 0;
+  showPrevArrow = false;
+  showNextArrow = true;
+  guideItemsLen: number;
+  visibleCards: number;
+
+  constructor(public guideItems: GuideItems) {
+    this.guideItemsLen = guideItems.getAllItems().length;
```

Comment:
nit: I would kind of prefer that `Length` is spelled out... But I won't fight this.